src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,40 @@
 'use strict';
-
 class Animal {
-  // write your code here
+  static alive = [];
+  static deleteIndex = Animal.alive.findIndex((animal) => animal.health <= 0);
+  constructor(health, name) {
+    this.health = 100;
+    this.name = name;
+    Animal.alive.push(this);
+  }
+
+  deleteAnimale() {
+    Animal.alive = Animal.alive.filter((animal) => animal.health > 0);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(health, name, hidden = false) {
+    super(health, name);
+    this.hidden = hidden;
+  }
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (animal.hidden === false) {
```

Comment:
The condition `animal.hidden === false` is not enough to ensure that the `bite` method only works on herbivores and not on carnivores. You need to add a check to determine if the `animal` is an instance of `Herbivore` before applying the bite effect.